encfs/encfssh

```diff
@@ -28,42 +29,63 @@ if [ ! -z "$2" ]; then
 	    mount_public=true
 	fi
     done
-    [ -d "$unenc_dir" ] || mkdir $unenc_dir
+    [ -d "$unenc_dir" ] || mkdir -- "$unenc_dir"
 else
     unenc_dir=$(mktemp -d .XXXXXXXX)
 fi
 
 if [ ! -d "$enc_dir" ]; then
-    mkdir $enc_dir
+    mkdir -- "$enc_dir"
 fi
 
 enc_dir=$(canonicalize "$enc_dir")
 unenc_dir=$(canonicalize "$unenc_dir")
 
-options=""
-if $unenc_dir_given; then
-    if $mount_public; then
-	options="-- -o allow_other"
+# clear command-line argument list; hereafter used for options
+set --
```

Comment:
From my perspective -- as someone trying to be certain of correctness -- I'm a lot happier when I don't have to do that analysis: Expanding an array is always certain never to perform glob expansions, unwanted string-splitting, etc; thus, when proofreading code, I don't need to look back and see what values can be present in a string when expanding it unquoted to get a similar effect.

That said, if `-- -o allow_other` are the only options we'll ever need to handle, and if we can be certain that `IFS` is set at its default value, then `$options` suffices.